apps/web/pages/settings/security/password.tsx

```diff
@@ -68,24 +113,55 @@ const PasswordView = () => {
         <Form form={formMethods} handleSubmit={handleSubmit}>
           <div className="max-w-[38rem] sm:flex sm:space-x-4">
             <div className="flex-grow">
-              <PasswordField {...register("oldPassword")} label={t("old_password")} />
+              <PasswordField {...formMethods.register("oldPassword")} label={t("old_password")} />
             </div>
             <div className="flex-grow">
-              <PasswordField {...register("newPassword")} label={t("new_password")} />
+              <PasswordField {...formMethods.register("newPassword")} label={t("new_password")} />
             </div>
           </div>
-          <p className="text-sm text-gray-600">
+          <p className="mt-4 max-w-[38rem] text-sm text-gray-600">
             <Trans i18nKey="invalid_password_hint">
               Password must be at least at least 7 characters, mix of uppercase & lowercase letters, and
-              contain at least 1 number
+              contain at least 1 number.
             </Trans>
           </p>
+          <div className="mt-8 border-t border-gray-200 py-8">
+            <SettingsToggle
+              title={t("session_timeout")}
+              description={t("session_timeout_description")}
+              checked={sessionTimeoutWatch !== undefined}
+              data-testid="session-check"
+              onCheckedChange={(e) => {
+                if (!e) {
+                  formMethods.setValue("sessionTimeout", undefined, { shouldDirty: true });
+                } else {
+                  formMethods.setValue("sessionTimeout", 10, { shouldDirty: true });
+                }
+              }}
+            />
+            {sessionTimeoutWatch && (
+              <div className="mt-4 text-sm">
+                <div className="flex items-center">
+                  <p className="text-neutral-900 ltr:mr-2 rtl:ml-2">{t("session_timeout_after")}</p>
+                  <Select
+                    options={timeoutOptions}
+                    defaultValue={timeoutOptions[1]}
```

Comment:
on a new load, the select always shows '10 mins' as the selected value and not the value that is actually selected

Reply:
Fixed! please try again, and thank you for your feedback 🙏